api/gateway.proto

```diff
@@ -127,8 +127,13 @@ message Gateway {
   google.protobuf.Duration schedule_anytime_delay = 19 [(gogoproto.stdduration) = true, (gogoproto.nullable) = true];
   // update the location of this gateway from status messages
   bool update_location_from_status = 21;
+  // An encypted secret value for this gateway.
+  // When set, this is currently sent as the lnsKey parameter in the Basic Station CUPS response. This can also be extended to store other values such as client TLS private key.
+  // This value cannot be set/retrieved in Gateway API calls. Use the StoreGatewaySecret, RetrieveGatewaySecret requests.
+  // It can however be deleted by DeleteGateway since it's part of the entity.
+  bytes secret = 22;
```

Comment:
I think this is too simple of a field and description.

I would suggest a `message Secrets` with repeated keyed secrets with a reference to the encryption key (or the encrypted DEK). This could be reusable for entities – I can imagine use cases for `Application` as well – or inline in `Gateway`.

See https://github.com/packetbroker/api/blob/master/v3/keys.proto for some inspiration.